File: Python/Program/Keyboard.py
from Periferal import Periferal

class Keyboard(Periferal):
    def __init__(self, tipe_switch="", is_rgb=False, jenis="", koneksi=""):
        super().__init__(jenis, koneksi)
        self.__tipe_switch = tipe_switch
        self.__is_rgb = is_rgb
    
    def set_tipe_switch(self, tipe_switch):
        self.__tipe_switch = tipe_switch
    
    def set_is_rgb(self, is_rgb):
        self.__is_rgb = is_rgb
    
    def get_tipe_switch(self):
        return self.__tipe_switch
    
    def get_is_rgb(self):
        return self.__is_rgb

File: CPP/Program/Keyboard.cpp
#ifndef KEYBOARD_CPP
#define KEYBOARD_CPP

#include <iostream>
#include <string>
#include "Periferal.cpp"
using namespace std;

class Keyboard : public Periferal {
private:
    string tipeSwitch; // Mechanical, Membrane, dll
    bool isRGB;
    
public:
    Keyboard() {
    }
    
    Keyboard(string tipeSwitch, bool isRGB, string jenis, string koneksi) : Periferal(jenis, koneksi) {
        this->tipeSwitch = tipeSwitch;
        this->isRGB = isRGB;
    }
    
    void setTipeSwitch(string tipeSwitch) {
        this->tipeSwitch = tipeSwitch;
    }
    
    void setIsRGB(bool isRGB) {
        this->isRGB = isRGB;
    }
    
    string getTipeSwitch() {
        return this->tipeSwitch;
    }
    
    bool getIsRGB() {
        return this->isRGB;
    }
    
    ~Keyboard() {
    }
};

#endif // KEYBOARD_CPP